turbo/snapshotsync/block_reader.go

```diff
@@ -498,7 +499,8 @@ func (back *BlockReaderWithSnapshots) BlockWithSenders(ctx context.Context, tx k
 }
 
 func (back *BlockReaderWithSnapshots) headerFromSnapshot(blockHeight uint64, sn *HeaderSegment, buf []byte) (*types.Header, []byte, error) {
-	if sn.idxHeaderHash == nil {
+	if sn == nil || sn.idxHeaderHash == nil {
+		log.Warn("missing header segment pieces run erigon with --download.verify")
```

Comment:
1. Actually absence of index it's a bit "valid case" - because we can't create all snapshot files atomically (it's not DB). So, there always will be small piece of time - when you have snapshots but don't have indices. 
`sapshots.ViewHeaders` does check that snapshots of given number exists (and opened), but doesn't check Index. 

2. I don't understand how this change can help in https://github.com/ledgerwatch/erigon/issues/5013 case. As I understand `sn.idxHeaderHash` is not nil (because this check already exists). And panic happened deeper.

3. About PR itself: I think better separate such 2 cases and add to warning message as much details as possible. 
then User can create for us more useful bug report.
Can add: s.from,  dbg.Stack(). 
But actually `sn == nil` is impossible because `ViewHeaders` method does check it.

Reply:
If this is the case then my PR loses purpose, so I will go ahead and close it.

Reply:
I mean - having panic is not ok. Return nil when indices are not ready yet - is ok.